File: leetcode/268.java
/*
Method 1
*/

class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int missing_number = n;

        for (int i = 0; i < n; i++) {
            missing_number ^= i ^ nums[i];
        }
        return missing_number;
    }
}

/*
Method 2
*/
/*
class Solution {
    public int missingNumber(int[] nums) {
        int n = nums.length;
        int sumArr = 0;
        for (int i = 0; i < n; i++) {
            sumArr += nums[i];
        }
        int seriesSum = n * (n + 1)/2;
        return seriesSum - sumArr;
    }
}
*/

File: solutions/268. Missing Number/268.py
class Solution:
    def missingNumber(self, nums: List[int]) -> int:
        sumArr = 0
        for i in range(len(nums)):
            sumArr += nums[i]

        n = len(nums)
        seriesSum = n * (n + 1)//2
        return seriesSum - sumArr
